src/python/pants/engine/scheduler.py

```diff
@@ -504,8 +505,31 @@ def product_request(self, product, subjects):
     :param list subjects: A list of subjects or Params instances for the request.
     :returns: A list of the requested products, with length match len(subjects).
     """
-    request = self.execution_request([product], subjects)
-    returns, throws = self.execute(request)
+    request = None
+    try:
+      request = self.execution_request([product], subjects)
+    except Exception:
```

Comment:
Swallowing this exception in the case that the engine had errors seems bad. Can we add it to the list of error tracebacks?

Reply:
Yes! The exception itself is however usually unrelated, e.g. something like:
> `No @rules were found to satisfy X for the input Select(Any+Y)`

since CFFI extern methods return None when the error is captured. I think it's safer to add these things and remove them later though, so I will add that to the end of the list of tracebacks, with some prefix like:
> `The engine execution request returned this error, which is probably because of the errors in the CFFI extern methods listed above.`

Reply:
Done!